actix-server/src/worker.rs

```diff
@@ -135,6 +135,13 @@ pub(crate) struct ServerWorker {
     shutdown_timeout: Duration,
 }
 
+/// Set worker to unavailable when dropping.
+impl Drop for ServerWorker {
+    fn drop(&mut self) {
+        self.rx.close();
```

Comment:
This already happens when the receiver is dropped, no?

Reply:
The drop order could be different than explicit call close and we want Accept to know the close asap. Like every cycle matters.

Reply:
the drop order is well defined in rust, rx is the first field and will be dropped first

Reply:
Future maintainer could be change the order and accidently make the time window larger.
And `UnboundedReciever` drop does not call close immediately. It does an atomic read first and return if there is no sender anymore. It's an optimization on the channel part but it make the close a bit later.

Reply:
yep, if you think it's important to make this explicit then we'll keep it

Reply:
I have added a note for ServerWorker field and removed the close in drop